gyp/pylib/gyp/input.py

```diff
@@ -16,9 +16,9 @@
 import sys
 import threading
 import traceback
-from distutils.version import StrictVersion
 from gyp.common import GypError
 from gyp.common import OrderedSet
+from packaging.version import Version
```

Comment:
Yes, the `packaging` dependency is run-time required here, not just in testing.

<details><summary>Error log (click to expand):</summary>

```
# ~/node-gyp/bin/node-gyp.js rebuild          
gyp info it worked if it ends with ok
gyp info using node-gyp@9.4.0
gyp info using node@16.20.2 | linux | x64
undefined
gyp info find Python using Python version 3.10.12 found at "/usr/bin/python3"
gyp info spawn /usr/bin/python3
gyp info spawn args [
gyp info spawn args '/root/node-gyp/gyp/gyp_main.py',
gyp info spawn args 'binding.gyp',
gyp info spawn args '-f',
gyp info spawn args 'make',
gyp info spawn args '-I',
gyp info spawn args '/root/require_superstring/node_modules/superstring/build/config.gypi',
gyp info spawn args '-I',
gyp info spawn args '/root/node-gyp/addon.gypi',
gyp info spawn args '-I',
gyp info spawn args '/root/.cache/node-gyp/16.20.2/include/node/common.gypi',
gyp info spawn args '-Dlibrary=shared_library',
gyp info spawn args '-Dvisibility=default',
gyp info spawn args '-Dnode_root_dir=/root/.cache/node-gyp/16.20.2',
gyp info spawn args '-Dnode_gyp_dir=/root/node-gyp',
gyp info spawn args '-Dnode_lib_file=/root/.cache/node-gyp/16.20.2/<(target_arch)/node.lib',
gyp info spawn args '-Dmodule_root_dir=/root/require_superstring/node_modules/superstring',
gyp info spawn args '-Dnode_engine=v8',
gyp info spawn args '--depth=.',
gyp info spawn args '--no-parallel',
gyp info spawn args '--generator-output',
gyp info spawn args 'build',
gyp info spawn args '-Goutput_dir=.'
gyp info spawn args ]
Traceback (most recent call last):
  File "/root/node-gyp/gyp/gyp_main.py", line 42, in <module>
    import gyp  # noqa: E402
  File "/root/node-gyp/gyp/pylib/gyp/__init__.py", line 9, in <module>
    import gyp.input
  File "/root/node-gyp/gyp/pylib/gyp/input.py", line 21, in <module>
    from packaging.version import Version
ModuleNotFoundError: No module named 'packaging'
gyp ERR! configure error 
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack at ChildProcess.onCpExit (/root/node-gyp/lib/configure.js:305:16)
gyp ERR! stack at ChildProcess.emit (node:events:513:28)
gyp ERR! stack at Process.ChildProcess._handle.onexit (node:internal/child_process:293:12)
gyp ERR! System Linux 5.15.49-linuxkit
gyp ERR! command "/usr/local/bin/node" "/root/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /root/require_superstring/node_modules/superstring
gyp ERR! node -v v16.20.2
gyp ERR! node-gyp -v v9.4.0
gyp ERR! not ok 
```

</details>